lib/ncr/work_order_reapproval_checker.rb

```diff
@@ -6,13 +6,21 @@ def initialize(work_order)
       @work_order = work_order
     end
 
-    def current_user
-      work_order.modifier
+    def requires_budget_reapproval?
+      work_order.approved? &&
+        work_order.requires_approval? &&
+        !budget_approver? &&
+        (amount_increased? || protected_fields_changed?)
     end
 
-    def previous_val(key)
-      changes = work_order.previous_changes || {}
-      changes[key.to_s].try(:first)
+    private
+
+    def budget_approver?
+      work_order.budget_approvers.include?(current_user)
+    end
```

Comment:
@afeld are you talking about the edge case where a delegate approves, then modifies? That is covered by the new definition of `completed_by` -- that could be the mailbox (`user`) or the delegate (`completer`) of the `step`.

That's a subset of `budget_mailxboxes_and_their_delegates` limited to the delegates who **actually clicked Approve**.